nav2_smac_planner/CMakeLists.txt

```diff
@@ -1,8 +1,6 @@
 cmake_minimum_required(VERSION 3.5)
 project(nav2_smac_planner)
 
-set(CMAKE_BUILD_TYPE Release) #Debug, Release
```

Comment:
We need this for the optimization that it provides, this package in particular is release-necessity for good performance

Reply:
Yes, but I believe that `nav2_common` already does this, in a nicer way:

https://github.com/ros-navigation/navigation2/blob/6bf9d68af4c485c11947114e0e91ea5b0b5efc27/nav2_common/cmake/nav2_package.cmake#L21-L28

Reply:
I think about it a bit differently, but perhaps you're right I shouldn't make multiple gates for folks to get through and know about to compile with debug if they want to